src/handlers/__tests__/displayNameHandler-test.js

```diff
@@ -70,17 +70,30 @@ describe('defaultPropsHandler', () => {
     expect(documentation.displayName).not.toBeDefined();
   });
 
-  it('ignores non-literal names', () => {
-    var definition = expression('({displayName: foo.bar})');
-    expect(() => displayNameHandler(documentation, definition)).not.toThrow();
-    expect(documentation.displayName).not.toBeDefined();
-
+  it('infers the displayName with es6 class', () => {
+    var definition = expression('({displayName: "FooBar"})');
+    displayNameHandler(documentation, definition);
+    expect(documentation.displayName).toBe('FooBar');
     definition = statement(`
       class Foo {
-        static displayName = foo.bar;
+
       }
     `);
-    expect(() => displayNameHandler(documentation, definition)).not.toThrow();
-    expect(documentation.displayName).not.toBeDefined();
+    displayNameHandler(documentation, definition);
+    expect(documentation.displayName).toBe('Foo');
+  });
+
+  it('infers the displayName with stateless functional component', () => {
+    var definition = expression('({displayName: "FooBar"})');
+    displayNameHandler(documentation, definition);
+    expect(documentation.displayName).toBe('FooBar');
+
+    definition = statement(`
+      var Foo = () => {
```

Comment:
Unfortunately this is not useful test, because the `displayNameHandler` function will never be passed a *VariableDeclaration*.

If you look at `getMemberValuePath`, you can see the node types which represent component definitions:

```
var LOOKUP_METHOD = {
  [types.ArrowFunctionExpression.name]: getMemberExpressionValuePath,
  [types.FunctionExpression.name]: getMemberExpressionValuePath,
  [types.FunctionDeclaration.name]: getMemberExpressionValuePath,
  [types.VariableDeclaration.name]: getMemberExpressionValuePath,
  [types.ObjectExpression.name]: getPropertyValuePath,
  [types.ClassDeclaration.name]: getClassMemberValuePath,
  [types.ClassExpression.name]: getClassMemberValuePath,
};
```

For some of them, inferring the name is easy: You can simple take the `id` of named classes and functions.

For all others, you need to traverse up the tree to find and assignment or export and inspect that one to get the name.

Here is how `getMemberExpressionValuePath` does it: https://github.com/reactjs/react-docgen/blob/9042c91d6b9e787f4fa49a96b7be0babb256fd3c/src/utils/getMemberExpressionValuePath.js#L36-L45